### ISSUE TITLE
Right Links WE stopped working

### ISSUE THREAD

**Issue by Flummoxed2** [Infocatcher/Right_Links_WE#27]
Stopped working for no reason. Firefox 87.0. Restarted Win10, no change. Thank you.

**Comment by Infocatcher** [comment]
With latest v.0.5b12?

Also please enable debug mode in the options and check out Error/Browser Console (Ctrl+Shift+J to open): should appears `[Right Links WE] …` messages and (probably) some errors/warnings (unfortunately, not so easy to investigate the source, URL will be like `moz-extension://<randomID>/<path>.js`).

**Comment by grammarbook** [comment]
Version 0.5b12 stopped working for me as well in the following environments:

- firefox 78.7.0esr (64-bit) under Linux
- firefox 88.0 (64 bit) under Windows 10

I do not see any output in the browser console, even with Debug mode enabled. I only use "Long left-click". Re-instllation of the add-on does not help.

**Comment by grammarbook** [comment]
Hmmm, sometimes, it does work, sometimes it does not. Couldn't figure out necessary / sufficient conditions for neither case.

**Comment by Infocatcher** [comment]
> I do not see any output in the browser console, even with Debug mode enabled. I only use "Long left-click". Re-instllation of the add-on does not help.

Strange… on startup should appears `setState(true)` message (or `setState(false)` in disabled state).
As I see, may fail only if something went wrong with `browser.storage.local.get()` browser API.

May be will help complete removal before re-installation (to reset options storage). But this is really strange anyway.

**Comment by chuvak2033** [comment]
Hi, today I noticed that a long left-click on a link does not work (for me it opens a link in the current tab). I don't know exactly when it stopped working, I felt that something was wrong, but I realized what was wrong only today)
I immediately looked in the reviews that this is not only me. I tried to uninstall / install - it did not help. Rolled back to version 0.5b9 - it worked. I installed the latest version again - it works. Then I adjusted all the other options and everything stopped working. Using the brute-force method, I found that if you uncheck the "Enabled Keyboard shortcut:" option, it stops working.
For me, this option is not important, but usually during setup I disable it and therefore a long left-click on a link stopped working for me.

**Comment by Infocatcher** [comment]
> Using the brute-force method, I found that if you uncheck the "Enabled Keyboard shortcut:" option, it stops working.

"Enabled" checkbox in the options – is a global enabled/disabled state, that will be displayed in toolbar button icon.
To disable keyboard shortcut – use empty string in text field.

**Comment by chuvak2033** [comment]
I got it, this option enables / disables the add-on, I'll put an empty value there. Before that there was а hot key F2 there, apparently I pressed it recently and everything stopped working. I slowpoke) Thought by mistake that this option was an enable / disable hotkey, thanks)

**Comment by grammarbook** [comment]
> I got it, this option enables / disables the add-on, I'll put an empty value there. Before that, there was а hot key F2 there, apparently I pressed it recently and everything stopped working. I slowpoke) Thought by mistake that this option was an enable / disable support hotkey, thanks)

May be I was mistaken in the same way. For me, it comes as a surprise that the "Enabled" checkbox does not refer to the keyboard shortcut but to the activation of the add-on. This misunderstanding is facilitated by the presence of the other on- / off-switch which is common to all add-ons, see the green ellipse in screenshot 1. In addition, these two switches are not kept in sync, see screenshot 2. Therefore, I'd recommend to remove the "Enabled" checkbox.
![screenshot_1](https://user-images.githubusercontent.com/61792881/120051101-9cac7880-c01f-11eb-8269-544eb6638849.jpg)
![screenshot_2](https://user-images.githubusercontent.com/61792881/120051110-a59d4a00-c01f-11eb-8377-46e8d0dba5be.png)

**Comment by Gitoffthelawn** [comment]
Another choice to improve the UI/UX is to add a checkbox to the left of the *"Keyboard shortcut"* string and make that checkbox the control to toggle the availability of the keyboard shortcut (vs having to provide an empty keyboard shortcut).

Then the *"Enabled"* string can either be removed or renamed to something more easily understood, perhaps *"Enable extension functionality"* or *"Extension functionality enabled"*.

**Comment by grammarbook** [comment]
I fully agree with the suggestion of Gitoffthelawn. By the way: Since I moved the Right Links WE icon (which also reflects the internal on/off state of the add-on) to the overflow menu of the Firefox toolbar, I also missed the second visual indicator of the add-on's internal state.
